index.bs

```diff
@@ -1923,6 +1923,34 @@ This section defines an initial set of extensions. These are recommended for imp
 interoperability.
 
 
+## FIDO AppId ## {#extension-appid}
+
+This authentication extension allows [RP]s who have previously registered an
+Authentication using the legacy FIDO JavaScript APIs to communicate with
+Authenticators.  Specifically, this extension allows [RP]s to specify an
+|appId| [[FIDO-APPID]] to overwrite the otherwise computed |rpId|.
+
+: Extension identifier
+:: `webauthn_appid`
+
+: Client argument
+:: A single UTF-8 encoded string specifying a FIDO |appId|.
+
+: Client processing
+:: The Client must perform AppId validation (as defined by [[FIDO-APPID]]).  If
+    valid, the value of |rpId| for all Client processing should be replaced by
+    the specified |appId|.
```

Comment:
What if rpId is also specified in the options? Should this win, or should the call just error out? (FWIW I'm in favor of erroring out by throwing an exception - it is a programming error and the code should never be written that way.)